src/actions/WindowActions.js

```diff
@@ -454,7 +457,29 @@ export function patch(
                 response.data, isModal, rowId, id, windowType, isAdvanced
             ));
 
-            dispatch(indicatorState('saved'));
+            // reinitialize layout if language is changed
```

Comment:
this is not needed. You just need to react to the websocket user session changed event.